java-checks-test-sources/src/main/java/checks/OSCommandsPath.java

```diff
@@ -0,0 +1,418 @@
+  private static final String[] COMPLIANT_COMMAND_ARRAY_UNIX_PARENT = new String[]{"../make"};
+  private static final String[] COMPLIANT_COMMAND_ARRAY_UNIX_CURRENT = new String[]{"./make"};
+  private static final String[] COMPLIANT_COMMAND_ARRAY_UNIX_HOME = new String[]{"~/bin/make"};
+  private static final String[] COMPLIANT_COMMAND_ARRAY_WINDOWS_ABSOLUTE = new String[]{"W:\\\\make"};
+  private static final String[] COMPLIANT_COMMAND_ARRAY_WINDOWS_PARENT = new String[]{"..\\bin\\make"};
+  private static final String[] COMPLIANT_COMMAND_ARRAY_WINDOWS_CURRENT = new String[]{".\\make"};
+  private static final String[] COMPLIANT_COMMAND_ARRAY_WINDOWS_NETWORK = new String[]{"\\\\SERVER\\make"};
+
+  private static final String[] ENVIRONMENT = new String[]{"DEBUG=true"};
+  private static final File FILE = null;
+
+  private static final List<String> NON_COMPLIANT_COMMAND_LIST = Arrays.asList("make");
+  private static final List<String> NON_COMPLIANT_COMMAND_LIST_UNIX_PARENT = Arrays.asList("m../ake");
+  private static final List<String> NON_COMPLIANT_COMMAND_LIST_UNIX_CURRENT = Arrays.asList("mak./e");
+  private static final List<String> NON_COMPLIANT_COMMAND_LIST_UNIX_HOME = Arrays.asList("bin~/make");
+  private static final List<String> NON_COMPLIANT_COMMAND_LIST_WINDOWS_ABSOLUTE = Arrays.asList("7:\\\\make");
+  private static final List<String> NON_COMPLIANT_COMMAND_LIST_WINDOWS_PARENT = Arrays.asList("m..\\ake");
+  private static final List<String> NON_COMPLIANT_COMMAND_LIST_WINDOWS_CURRENT = Arrays.asList("ma.\\ke");
+  private static final List<String> NON_COMPLIANT_COMMAND_LIST_WINDOWS_NETWORK = Arrays.asList("SERVER\\make");
+  private static final List<String> NON_COMPLIANT_COMMAND_LIST_VARIABLE = Arrays.asList(NON_COMPLIANT_COMMAND);
+
+  private static final List<String> COMPLIANT_COMMAND_LIST_UNIX = Arrays.asList("/usr/bin/make");
+  private static final List<String> COMPLIANT_COMMAND_LIST_UNIX_PARENT = Arrays.asList("../make");
+  private static final List<String> COMPLIANT_COMMAND_LIST_UNIX_CURRENT = Arrays.asList("./make");
+  private static final List<String> COMPLIANT_COMMAND_LIST_UNIX_HOME = Arrays.asList("~/bin/make");
+  private static final List<String> COMPLIANT_COMMAND_LIST_WINDOWS_ABSOLUTE = Arrays.asList("Z:\\make");
+  private static final List<String> COMPLIANT_COMMAND_LIST_WINDOWS_PARENT = Arrays.asList("..\\make");
+  private static final List<String> COMPLIANT_COMMAND_LIST_WINDOWS_CURRENT = Arrays.asList(".\\make");
+  private static final List<String> COMPLIANT_COMMAND_LIST_WINDOWS_NETWORK = Arrays.asList("\\\\SERVER\\make");
+  private static final List<String> COMPLIANT_COMMAND_LIST_VARIABLE = Arrays.asList(File.pathSeparator);
+
+  private static final String UNINITIALIZED_COMMAND = null;
+  private static final String[] UNINITIALIZED_COMMAND_ARRAY = null;
+  private static final List<String> UNINITIALIZED_COMMAND_LIST = null;
+
+  public void falseNegatives() throws IOException {
```

Comment:
I would rather put this method later in the file, because it would make more sense to have the cases where the rule reports an issue first, before having to understand the cases where the rule does not report an issue.